bot/services/paginate_service.py

```diff
@@ -0,0 +1,105 @@
+                                    channel: discord.TextChannel,
+                                    timeout: int = 60):
+
+        if not isinstance(pages , t.List):
+            pages = [pages]
+
+        embed = discord.Embed(title= embed_name, color= Colors.ClemsonOrange)
+        # set the first page of the embed
+        embed.add_field(name= field_title, value= pages[0])
+        embed.set_footer(text=f'Page 1 of {len(pages)}')
+        msg = await channel.send(embed= embed)
+        await self.bot.messenger.publish(Events.on_set_deletable, msg=msg, author=msg.author)
+
+        # stores the message info
+        self.messages[msg.id] = {
+            "embed_name": embed_name,
+            "field_title": field_title,
+            "curr_page": 0,
+            "pages": pages,
+            "author": author.id if author else None
+        }
+
+        # add every emoji from the reaction list
+        for reaction in self.reactions:
+            await msg.add_reaction(reaction)
+        if timeout:
+            await asyncio.sleep(timeout) 
+            try:
+                for reaction in self.reactions:
+                    await msg.clear_reaction(reaction)
+                del self.messages[msg.id]
+            except:
+                pass
+            finally:
+                log.info(f'Message: {msg.id} timed out as pageable')
+
+    @BaseService.Listener(Events.on_reaction_add)
+    async def change_page(self, reaction: discord.Reaction, user: t.Union[discord.User, discord.Member]):
+        perm = False
+
+        if reaction.emoji not in self.reactions or reaction.message.id not in self.messages:
+            return
+        elif user.guild_permissions.administrator:
+            perm = True
+        elif user.id == self.messages[reaction.message.id]["author"]:
+            perm = True
+
+        # check if user has perm to change the page
+        if perm:
+            msg = self.messages[reaction.message.id]
+            embed = discord.Embed(title =  msg["embed_name"], color=Colors.ClemsonOrange)
+            # check what emoji user used
+            if reaction.emoji == "⏮️":
+                if msg["curr_page"] != 0:
+                    self.messages[reaction.message.id]["curr_page"] = 0
+            elif reaction.emoji == "⬅️":
+                if msg["curr_page"] != 0:
+                    self.messages[reaction.message.id]["curr_page"] -= 1
+            elif reaction.emoji == "➡️":
+                if msg["curr_page"] < len(msg["pages"])-1:
+                    self.messages[reaction.message.id]["curr_page"] += 1
+            elif reaction.emoji == "⏭️":
+                if msg["curr_page"] != len(msg["pages"])-1:
+                    self.messages[reaction.message.id]["curr_page"] = len(msg["pages"])-1
+
+            # edits the message and reset emoji (only user that triggered the action)
+            embed.add_field(name= msg["field_title"], value=msg["pages"][msg["curr_page"]])
```

Comment:
if you wanted to you could make the messages dictionary into an `@dataclass` object, that would all you to define a getter for the current page content to avoid having to do this `msg["pages"][msg["curr_page"]]`